syntax = "proto3";

package protos;

service ObjectTracking{
 rpc track_objects(Request) returns (TrackingResponse) {}
}

message Request{
 string image = 1;
 int64 frame_id = 2;
 int64 request_time_ms = 3;
}

message TrackingResponse{
 int64 frame_id = 1;
 int64 request_time_ms = 2;
 int64 request_received_time_ms = 3;
 int64 response_time_ms = 4;
 repeated TrackedObject tracked_objects = 5;
}

message TrackedObject {
  int32 track_id = 1;
  string clazz = 2;
  int32 x_min = 3;
  int32 x_max = 4;
  int32 y_min = 5;
  int32 y_max = 6;
}
